caluma/form/tests/test_validators.py

```diff
@@ -44,77 +44,6 @@ def test_validate_special_fields(
     DocumentValidator().validate(document, info)
 
 
-@pytest.mark.parametrize(
-    "main_required,required_jexl,should_throw",
-    [
-        ("true", "true", True),
-        ("true", "false", False),
-        ("true", "'parent.sub_2.sub_2_question_1'|answer == 'foo'", True),
-        ("true", "'parent.sub_2.sub_2_question_1'|answer == 'bar'", False),
-        ("false", "true", False),
-        ("false", "false", False),
-        ("false", "'parent.sub_2.sub_2_question_1'|answer == 'foo'", False),
-        ("false", "'parent.sub_2.sub_2_question_1'|answer == 'bar'", False),
-    ],
-)
-def test_validate_nested_form(
-    db,
-    required_jexl,
-    should_throw,
-    main_required,
-    form,
-    form_question_factory,
-    document_factory,
-    question_factory,
-    answer_factory,
-    answer_document_factory,
-    info,
-):
-    sub_form_question_1 = form_question_factory(
-        form__slug="sub_1",
-        question__type=Question.TYPE_TEXT,
-        question__is_required=required_jexl,
-        question__slug="sub_1_question_1",
-    )
-    sub_form_question_2 = form_question_factory(
-        form__slug="sub_2",
-        question__type=Question.TYPE_TEXT,
-        question__is_required="true",
-        question__slug="sub_2_question_1",
-    )
-
-    main_form_question_1 = form_question_factory(
-        question__type=Question.TYPE_FORM,
-        question__sub_form=sub_form_question_1.form,
-        question__slug="sub_1",
-        question__is_required=main_required,
-    )
-    form_question_factory(
-        form=main_form_question_1.form,
-        question__type=Question.TYPE_FORM,
-        question__sub_form=sub_form_question_2.form,
-        question__slug="sub_2",
-    )
-
-    main_document = document_factory(form=main_form_question_1.form)
-
-    Document.objects.create_and_link_child_documents(
-        main_form_question_1.form, main_document
-    )
-
-    sub_2_document = Document.objects.filter(form__slug="sub_2").first()
-    answer_factory(
-        document=sub_2_document, question=sub_form_question_2.question, value="foo"
-    )
-
-    if should_throw:
-        error_msg = f"Questions {sub_form_question_1.question.slug} are required but not provided"
-        with pytest.raises(ValidationError, match=error_msg):
-            DocumentValidator().validate(main_document, info)
-    else:
-        DocumentValidator().validate(main_document, info)
```

Comment:
Shouldn't we adapt this test case to something that is still meaningful with the refactor? After all we can still have nested forms, just not nested answer trees.

Reply:
This tests the DocumentValidator() which does not have to handle nesting.

Reply:
Done!